registry.ts

```diff
@@ -0,0 +1,14 @@
+const modules: Record<string, unknown> = {};
+
+export const getModule = (name: string) => {
```

Comment:
Also, minor TS thing, it might be nice if you could give `getModule` a type so that the object it returns is actually typed. Unfortunately, I don't know if we can do anything fancy like automatically inferring the return type based on the given name, but it would be nice.

I'm thinking it could look something like Unity's `GetComponent` method
```c#
HingeJoint hinge = gameObject.GetComponent<HingeJoint>();
```

Reply:
I'd prefer it to be loosely typed so that we don't have to adhere to any specific type constraint wrt. React componentry or other. Since we are wanting to override arbitrary modules, I don't think there's any value in trying to type them.

Reply:
I'd think that's even more reason to be strict on the typing. If you can swap out modules, I'd want to ensure they have a common interface to reduce the chances that we break stuff accidentally